catalog/transformer/catalog-transformer-xlsx/src/main/java/ddf/catalog/transformer/xlsx/XlsxMetacardUtility.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.xlsx;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.BinaryContent;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.BinaryContentImpl;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellStyle;
+import org.apache.poi.ss.usermodel.Font;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class XlsxMetacardUtility {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(XlsxMetacardUtility.class);
+
+  private static MimeType mimeType = new MimeType();
+
+  static {
+    try {
+      mimeType.setPrimaryType("application");
+      mimeType.setSubType("vnd.openxmlformats-officedocument.spreadsheetml.sheet");
+    } catch (MimeTypeParseException e) {
+      LOGGER.warn("Failure creating MIME type", e);
+      throw new ExceptionInInitializerError(e);
+    }
+  }
+
+  public BinaryContent buildSpreadSheet(List<Metacard> metacards) {
+
+    Set<AttributeDescriptor> attributeDescriptors =
+        metacards.get(0).getMetacardType().getAttributeDescriptors();
```

Comment:
❗️ This could throw an `IndexOutOfBoundsException` if the list of metacards is empty. Your clients are checking for this however some other client may not be as wise.